Marlin/Marlin_main.cpp

```diff
@@ -5172,8 +5178,14 @@ void home_all_axes() { gcode_G28(true); }
         return;
       }
 
+      const int8_t force_iterations = parser.seen('F') ? parser.value_int() : 1;
+      if (!WITHIN(force_iterations, 1, 30)) {
+        SERIAL_PROTOCOLLNPGM("?(F)orce iteration is implausible (1-30).");
+        return;
+      }
+
       const bool towers_set = !parser.seen('T'),
-                 stow_after_each      = parser.seen('E') && parser.value_bool(),
+                 stow_after_each      = parser.seen('E'),
```

Comment:
`parser.value_bool()` is appropriate for all G-code parameters that act like flags. In these G-codes, the letter by itself (`G33 E`) is the same as the letter with a non-zero value, while the letter with zero (`G33 E0`) is another way (besides leaving off the parameter) of explicitly saying "false" to a given parameter. Using a bool parameter with "`0`" is often desirable, because the default behavior for a flag may be on or off, even differing between firmwares. Being able to specify `E0` to say "No Stow" is therefore an important capability.